src/sentry/api/helpers/group_index.py

```diff
@@ -207,3 +223,98 @@ def validate(self, attrs):
             raise serializers.ValidationError(
                 'Other attributes cannot be updated when discarding')
         return attrs
+
+
+class GroupIndexMixin(object):
+    def handle_discard(self, request, group_list, projects, user):
+        for project in projects:
+            if not features.has('projects:discard-groups', project, actor=user):
+                return Response({'detail': ['You do not have that feature enabled']}, status=400)
+
+        # grouped by project_id
+        groups_to_delete = defaultdict(list)
+
+        for group in group_list:
+            with transaction.atomic():
+                try:
+                    tombstone = GroupTombstone.objects.create(
+                        previous_group_id=group.id,
+                        actor_id=user.id if user else None,
+                        **{name: getattr(group, name) for name in TOMBSTONE_FIELDS_FROM_GROUP}
+                    )
+                except IntegrityError:
+                    # in this case, a tombstone has already been created
+                    # for a group, so no hash updates are necessary
+                    pass
+                else:
+                    groups_to_delete[group.project_id].append(group)
+
+                    GroupHash.objects.filter(
+                        group=group,
+                    ).update(
+                        group=None,
+                        group_tombstone_id=tombstone.id,
+                    )
+
+        for project in projects:
+            to_delete = groups_to_delete.get(project.id)
+            if to_delete:
+                self._delete_groups(request, project, to_delete, delete_type='discard')
+
+        return Response(status=204)
+
+    def _delete_groups(self, request, project, group_list, delete_type):
```

Comment:
didn't make this support multiple projects because eventstream doesn't support deleting groups from multiple projects